static/app/views/starfish/modules/databaseModule/panel.tsx

```diff
@@ -155,29 +176,59 @@ function QueryDetailBody({
       tableData.reduce((acc, transaction) => acc + transaction.p75, 0) / tableData.length;
   }
 
-  const mergedTableData = values(
-    merge(keyBy(eventCountData, 'transaction'), keyBy(tableData, 'transaction'))
-  ).filter((data: Partial<TransactionListDataRow>) => !!data.count && !!data.p75);
+  const eventCountMap = keyBy(eventCountData, 'transaction');
+
+  const mergedTableData: TransactionListDataRow[] = tableData.map(data => {
+    const {transaction} = data;
+    const eventData = eventCountMap[transaction];
+    if (eventData) {
+      return {...data, ...eventData};
+    }
+    return data;
+  });
 
   const [countSeries, p75Series] = throughputQueryToChartData(
     graphData,
     startTime,
     endTime
   );
 
-  function renderHeadCell(column: GridColumnHeader): React.ReactNode {
-    return <span>{column.name}</span>;
+  const onSortClick = (col: TableColumnHeader) => {
+    let direction: 'desc' | 'asc' | undefined = undefined;
+    if (sort.direction === 'desc') {
+      direction = 'asc';
+    } else if (!sort.direction) {
+      direction = 'desc';
+    }
+    setSort({direction, sortHeader: col});
+  };
+
+  function renderHeadCell(col: TableColumnHeader): React.ReactNode {
+    const {key, name} = col;
+    const sortableKeys: Keys[] = ['p75', 'count'];
```

Comment:
Should we allow sorting by frequency and total events too?

Reply:
Yeah we should, but because we're meshing two queries together, we'll have to make additonal changes for it to work properly